pkg/cache/cache.go

```diff
@@ -0,0 +1,90 @@
+package cache
+
+import (
+	"path"
+	"strings"
+
+	zlog "github.com/anuvu/zot/pkg/log"
+	"github.com/boltdb/bolt"
+)
+
+const (
+	BlobsCache = "blobs"
+)
+
+type Cache struct {
+	db  *bolt.DB
+	log zlog.Logger
+}
+
+// Blob is a blob record
+type Blob struct {
+	Path string
+}
+
+func NewCache(rootDir string, name string, log zlog.Logger) *Cache {
+	db, err := bolt.Open(path.Join(rootDir, name+".db"), 0600, nil)
+	if err != nil {
+		panic(err)
+	}
+
+	if err := db.Update(func(tx *bolt.Tx) error {
+		_, err := tx.CreateBucketIfNotExists([]byte(BlobsCache))
+		return err
+	}); err != nil {
+		panic(err)
+	}
+
+	return &Cache{db: db, log: log}
+}
+
+func (c *Cache) PutBlob(digest string, path string) {
+	if err := c.db.Update(func(tx *bolt.Tx) error {
+		root := tx.Bucket([]byte(BlobsCache))
+		if root == nil {
+			panic("")
```

Comment:
Seems like we should say *something* useful here :)

Reply:
Left some panic()s in coz that is assured to not happen as per the package. But safeguards us in case package versions change behavior.